daemon/errors.go

```diff
@@ -160,6 +160,17 @@ func setExitCodeFromError(setExitCode func(exitStatus), err error) error {
 		return startInvalidConfigError(errDesc)
 	}
 
+	// Go 1.20 changed the error for attempting to execute a directory from
+	// syscall.EACCESS to syscall.EISDIR. Unfortunately docker/cli checks
```

Comment:
This is a compatibility issue which manifests when _runc_ is compiled against Go 1.20, irrespective of the Go version that Moby is compiled with.